snapcraft/cli/_errors.py

```diff
@@ -58,35 +72,37 @@ def exception_handler(exception_type, exception, exception_traceback, *,
         traceback.print_exception(
             exception_type, exception, exception_traceback)
 
-    if not is_snapcraft_error:
-        echo.error('Sorry, Snapcraft had an internal error.')
-        if _is_send_error_data():
+    if RavenClient is not None and not is_snapcraft_error:
+        is_env_send_data = os.environ.get(
+            'SNAPCRAFT_SEND_ERROR_DATA', 'n') == 'y'
+        msg = _MSG_SEND_TO_SENTRY_TRACEBACK_CONFIRM
+        click.echo(_MSG_SEND_TO_SENTRY_TRACEBACK)
+        traceback.print_exception(
+            exception_type, exception, exception_traceback)
+        if is_env_send_data:
+            click.echo(_MSG_SEND_TO_SENTRY_ENV)
             _submit_trace(exception)
+            click.echo(_MSG_SEND_TO_SENTRY_THANKS)
+        elif click.confirm(msg):
+            _submit_trace(exception)
+            click.echo(_MSG_SEND_TO_SENTRY_THANKS)
+    else:
```

Comment:
Knowing that RavenClient is part of the conditional above, consider `elif is_snapcraft_error` here instead.